src/webxdc.rs

```diff
@@ -414,14 +419,15 @@ impl Context {
     }
 
     /// Attempts to send queued webxdc status updates.
-    pub(crate) async fn flush_status_updates(&self) -> Result<()> {
-        loop {
-            let (instance_id, first_serial, last_serial, descr) =
-                match self.pop_smtp_status_update().await? {
-                    Some(res) => res,
-                    None => return Ok(()),
-                };
+    ///
+    /// Returns true if there are more status updates to send, but rate limiter does not
+    /// allow to send them. Returns false if there are no more status updates to send.
+    pub(crate) async fn flush_status_updates(&self) -> Result<bool> {
```

Comment:
this needs to be rebased-or-so. `flush_status_updates()` comment and return value was changed in https://github.com/deltachat/deltachat-core-rust/pull/3597 - this reverts these changes.

apart from that, with or without #3597, the diff seems to be superfluous as after the loop always all instances are processed and removed from table afaik.

Reply:
> apart from that, with or without https://github.com/deltachat/deltachat-core-rust/pull/3597, the diff seems to be superfluous as after the loop always all instances are processed and removed from table afaik.

Yeah the diff is always empty and removing it from an existing set is thus a noop. At the time I wrote this I wanted to make it proof against future changes where not all update messages were sent out for whatever reason. Because it is a noop it semt fine to do that. With the new implementation that doesn't even send a `WebxdcUpToDate` event this is however rented useless anyway and will probably be removed